substrate-processor/src/interfaces/hooks.ts

```diff
@@ -3,36 +3,76 @@ import {EvmLogHandler, EvmTopicSet} from "./evm"
 import {BlockHandler, EventHandler, ExtrinsicHandler} from "./handlerContext"
 import {QualifiedName} from "./substrate"
 
-
+/**
+ * Defines a {@link BlockHandler} function and the {@link Range} of blocks that should restrain its execution
+ * 
+ * @property handler: {@link BlockHandler}
+ * @property range: {@link Range} (optional)
+ */
 export interface BlockHook {
     handler: BlockHandler
     range?: Range
 }
 
-
+/**
+ * Defines a {@link EventHandler} function, the {@link Range} of blocks, and the Event name that should restrain its execution
+ * 
+ * @property handler: {@link EventHandler}
+ * @property event: {@link QualifiedName} the name of the event that should trigger the handler
+ * @property range: {@link Range} (optional)
+ */
 export interface EventHook {
     handler: EventHandler
     event: QualifiedName
     range?: Range
 }
 
-
+/**
+ * Defines a {@link ExtrinsicHandler} function and the {@link Range} of blocks, Event and Extrinsic names that should restrain its execution
+ * 
+ * @property handler: {@link ExtrinsicHandler}
+ * @property event: {@link QualifiedName} the name of the event that should trigger the handler
+ * @property extrinsic: {@link QualifiedName} the name of the extrinsic that should trigger the handler
+ * @property range: {@link Range} (optional)
+ */
 export interface ExtrinsicHook {
     handler: ExtrinsicHandler
     event: QualifiedName
     extrinsic: QualifiedName
     range?: Range
 }
 
-
+/**
+ * Defines a {@link EvmLogHandler} function and the contract address that should restrain its execution.
+ * 
+ * Optionally, a {@link Range} of blocks and EvmTopicSet to further filter the function triggering can be provided.
+ * 
+ * @property handler: {@link EvmLogHandler}
+ * @property contractAddress: `string` representing the contract address that should trigger the handler
+ * @property filter: {@link EvmTopicSet} (otpional)
+ * @property range: {@link Range} (optional)
+ */
 export interface EvmLogHook {
     handler: EvmLogHandler
     contractAddress: string
     filter?: EvmTopicSet[]
     range?: Range
 }
 
-
+/**
```

Comment:
Here it should be noted that for each block the execution order is as follows:
- pre block hooks for this blocks
- event hooks are executed in the order the events are emitted within the block
- extrinsic hooks are executed in the order the triggerEvent ( `system.ExtrinsicSuccess` by default) are emitted within the block
- post block hooks are executed after all other hooks
- postHooks are executed after all the event and extrinsics in the block